File: backend/models/sms_parser.py
import xml.etree.ElementTree as ET

def parse_sms(file_path):
    """Reads an XML file and extracts SMS message details."""
    try:
        tree = ET.parse(file_path)  # Load and parse XML file
        root = tree.getroot()  # Get the root element

        sms_list = []  # Store parsed messages

        for sms in root.findall("sms"):  # Loop through all SMS entries
            sms_data = {
                "body": sms.attrib.get("body", "").strip(),
                "date": sms.attrib.get("date", ""),
            }
            sms_list.append(sms_data)  # Add to list

        return sms_list  # Return extracted SMS messages

    except Exception as e:
        print(f"Error parsing XML file: {e}")
        return []

# Example usage:
sms_data = parse_sms("../sms_data/sms.xml")
#debugging
print(sms_data[:5])  # Show first few parsed messages

File: backend/models/sms_parser.js
const { XMLParser } = require("fast-xml-parser");
const fs = require("fs");

function parseSMSFromFile(filePath) {
    try {
        const xmlData = fs.readFileSync(filePath, "utf-8");
        const parser = new XMLParser({ ignoreAttributes: false, attributeNamePrefix: "" });
        const result = parser.parse(xmlData);

        const smsArray = result.smses.sms || [];
        const smsList = Array.isArray(smsArray) ? smsArray : [smsArray];

        return smsList.map(sms => ({
            body: (sms.body || "").trim(),
            date: sms.date || ""
        }));
    } catch (e) {
        console.error("Error parsing XML file:", e);
        return [];
    }
}

// Example usage
const smsList = parseSMSFromFile("../../sms_data/sms.xml");
console.log(smsList);
